core/templates/dev/head/services/exploration-features-backend-api.service.ts

```diff
@@ -17,28 +17,37 @@
  * configured to support.
  */
 
-require('services/services.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('ExplorationFeaturesBackendApiService', [
-  '$http', 'UrlInterpolationService', 'EXPLORATION_FEATURES_URL',
-  function($http, UrlInterpolationService, EXPLORATION_FEATURES_URL) {
-    return {
-      /**
-       * Retrieves data regarding the features the given exploration supports.
-       *
-       * NOTE: This service requires play-access for the Exploration so that the
-       * features can be fetched in both the exploration player and editor.
-       *
-       * @returns {Object.<string, *>} - Describes the features the given
-       *     exploration supports.
-       */
-      fetchExplorationFeatures: function(explorationId) {
-        return $http.get(
-          UrlInterpolationService.interpolateUrl(
-            EXPLORATION_FEATURES_URL, {exploration_id: explorationId})
-        ).then(function(response) {
-          return response.data;
-        });
-      },
-    };
-  }]);
+import { ServicesConstants } from 'services/services.constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationFeaturesBackendApiService {
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private http: HttpClient
+  ) {}
+
+  _fetchExplorationFeatures(explorationId: String): Promise<Object> {
+    return this.http.get(
+      this.urlInterpolationService.interpolateUrl(
+        ServicesConstants.EXPLORATION_FEATURES_URL,
+        {exploration_id: explorationId}
+      )
+    ).toPromise();
```

Comment:
Just curious, we are returning promise here -- but if you see the previous code above, we are returning the data by resolving the promise. How is this working fine?